scalameta/common/shared/src/main/scala/scala/meta/internal/ast/ast.scala

```diff
@@ -172,6 +172,8 @@ class AstNamerMacros(val c: Context) extends AstReflection with CommonNamerMacro
       pelClauses += cq"_ => throw new $IndexOutOfBoundsException(n.toString)"
       stats1 += q"override def productElement(n: $IntClass): Any = n match { case ..$pelClauses }"
       stats1 += q"override def productIterator: $IteratorClass[$AnyClass] = $ScalaRunTimeModule.typedProductIterator(this)"
+      val productFields = productParamss.head.map(_.name.toString)
+      stats1 += q"override def productFields: $ListClass[$StringClass] = List(..$productFields)"
```

Comment:
I don't have a test suite at the top of my head. You can try to use it from the js parsers facade  ;)